app/services/session.js

```diff
@@ -25,4 +26,9 @@ export default class ExtendedSessionService extends SessionService {
       this.router.transitionTo('login');
     }
   }
+
+  async invalidate() {
+    await this.impersonation.stopImpersonation();
```

Comment:
My understanding of the auth addon we use is that `handleInvalidation` only gets called if the underlying session is no longer valid (maybe it expires while the user is logged in). If the user manually logs out, we never go through `handleInvalidation`. I will double check though, because that was last week and perhaps I'm just mistaken.